### PR TITLE
Bugfix - Also handle static:: in self-provider.

### PR DESCRIPTION
**Pull request by hotoiledgoblinsack**
Hello

The `static` keyword can be used to apply late static binding of static methods, which the `self` keyword does not apply. If I understand correctly (correct me if I'm wrong ;-), as the child class can be anything from the perspective of the class using the `static` keyword, it should probably be treated the same as the `self` keyword.

### CODE REVIEW

**Comment by Peekmo** [comment]
Hello,

See https://github.com/Peekmo/atom-autocomplete-php/pull/38. It seems that static:: can act like $this. So, I'm not sure that it's really the same as self::

**Comment by hotoiledgoblinsack** [comment]
It turns out you are right, from what I've gathered:
  * In a *static* method, **static** should behave like **self** (with added late-static binding, but this does not matter for us, I think).
  * In a *non-static* method, **static** should behave like **$this**, in the sense that you can call non-static class methods with the keyword.
  * In a *non-static* method, **self** also behaves like **$this**, in the same way as **static** does.

More concretely, all of these are *valid* PHP code:
```
// A non-static method.
public function testStaticMethodFromThisContext()
{
    $this->someClassMethodThatIsNotStatic();
    self::someClassMethodThatIsNotStatic();
    static::someClassMethodThatIsNotStatic();
}
```

Whilst all of these are *invalid* PHP code:

```
// A static method.
public static function testStaticMethodFromStaticContext()
{
    $this->someClassMethodThatIsNotStatic();
    self::someClassMethodThatIsNotStatic();
    static::someClassMethodThatIsNotStatic();
}
```

In short what I've added in this pull request is only partially correct. Seeing as there is more work to do for both the **self** and **static** keywords, perhaps the remaining work should be split off into a ticket (the code will need to be expanded to know whether the currenct completion context is a static method or not).
